website/www/site/content/en/documentation/io/built-in/snowflake.md

```diff
@@ -0,0 +1,738 @@
+
+  @Description("Schema to use. Optional.")
+  @Default.String("")
+  ValueProvider<String> getSchema();
+
+  void setSchema(ValueProvider<String> schema);
+
+  @Description("Table to use. Optional.")
+  @Default.String("")
+  ValueProvider<String> getTable();
+
+  void setTable(ValueProvider<String> table);
+
+  @Description("Query to use. Optional.")
+  @Default.String("")
+  ValueProvider<String> getQuery();
+
+  void setQuery(ValueProvider<String> query);
+
+  @Description("Role to use. Optional.")
+  @Default.String("")
+  ValueProvider<String> getRole();
+
+  void setRole(ValueProvider<String> role);
+
+  @Description("Authenticator to use. Optional.")
+  @Default.String("")
+  String getAuthenticator();
+
+  void setAuthenticator(String authenticator);
+
+  @Description("Port number. Optional.")
+  @Default.String("")
+  String getPortNumber();
+
+  void setPortNumber(String portNumber);
+
+  @Description("Login timeout. Optional.")
+  @Default.String("")
+  String getLoginTimeout();
+
+  void setLoginTimeout(String loginTimeout);
+
+  @Description("Temporary GCS bucket name.")
+  ValueProvider<String> getStagingBucketName();
+
+  void setStagingBucketName(ValueProvider<String> stagingBucketName);
+
+  @Description("Storage integration name")
+  ValueProvider<String> getStorageIntegrationName();
+
+  void setStorageIntegrationName(ValueProvider<String> storageIntegrationName);
+
+  @Description("SnowPipe name. Optional.")
+  ValueProvider<String> getSnowPipe();
+
+  void setSnowPipe(ValueProvider<String> snowPipe);
+}
+{{< /highlight >}}
+### Using Pipeline options
+To use Pipeline options, you must configure them as follows:
+
+{{< highlight java>}}
+SnowflakePipelineOptions options = PipelineOptionsFactory
+    .fromArgs(args)
+    .withValidation()
+    .as(SnowflakePipelineOptions.class);
+{{< /highlight >}}
+
+All the below parameters are required:
+- `.fromArgs()`
+  - GNU style command line arguments, 
+  - Example: `--project=myproject --database=test`
+- `.withValidation()`
+  - which validates that PipelineOptions confirms all criteria from the passed in interface
+- `.as()`
+  - a class of used pipeline options
+  - Example: `.as(SnowflakePipelineOptions.class)`
+
+Then create your pipeline using created options:
+
+{{< highlight java >}}
+Pipeline pipeline = Pipeline.create(options);
+{{< /highlight >}}
+
+Example of accessing pipeline options in code:
+
+{{< highlight java >}}
+String stagingBucketName = options.getStagingBucketName();
+{{< /highlight >}}
+
+Pipelines in tests:
+{{< highlight java >}}
+PipelineOptionsFactory.register(SnowflakePipelineOptions.class);
+    options = TestPipeline
+                .testingPipelineOptions()                                  
+                .as(SnowflakePipelineOptions.class);
+{{< /highlight >}}
+
+Then use them in pipeline run:
+{{< highlight java >}}
+PipelineResult pipelineResult = pipeline.run(options);
+{{< /highlight >}}
+
+### Extending pipeline options
```

Comment:
I would remove also this because Beam users know how to use PipelineOptions

Reply:
Removed last time